src/pages/workspaces/workspace/Dashboard/Dashboard.ts

```diff
@@ -0,0 +1,511 @@
+import { canEditWorkspace, canWrite, isGoogleWorkspace, isOwner } from 'src/libs/workspace-utils';
+import SignIn from 'src/pages/SignIn';
+import { CloudInformation } from 'src/pages/workspaces/workspace/Dashboard/CloudInformation';
+import { DataUseLimitations, displayAttributeValue } from 'src/pages/workspaces/workspace/Dashboard/DataUseLimitations';
+import { OwnerNotice } from 'src/pages/workspaces/workspace/Dashboard/OwnerNotice';
+import { RightBoxSection } from 'src/pages/workspaces/workspace/Dashboard/RightBoxSection';
+import { WorkspaceInformation } from 'src/pages/workspaces/workspace/Dashboard/WorkspaceInformation';
+import { WorkspaceNotifications } from 'src/pages/workspaces/workspace/Dashboard/WorkspaceNotifications';
+import DashboardPublic from 'src/pages/workspaces/workspace/DashboardPublic';
+import { displayLibraryAttributes } from 'src/pages/workspaces/workspace/library-attributes';
+import { InitializedWorkspaceWrapper as Workspace, StorageDetails } from 'src/pages/workspaces/workspace/useWorkspace';
+import { wrapWorkspace } from 'src/pages/workspaces/workspace/WorkspaceContainer';
+
+import { WorkspaceAcl } from '../WorkspaceAcl';
+
+const styles: Record<string, CSSProperties> = {
+  authDomain: {
+    padding: '0.5rem 0.25rem',
+    marginBottom: '0.25rem',
+    backgroundColor: colors.dark(0.15),
+    ...Style.noWrapEllipsis,
+  },
+  tag: {
+    padding: '0.25rem',
+    margin: '0.15rem',
+    backgroundColor: colors.dark(0.15),
+    borderRadius: 10,
+    overflow: 'hidden',
+    wordWrap: 'break-word',
+  },
+};
+
+const DashboardAuthContainer = (props: WorkspaceDashboardProps): ReactNode => {
+  const { namespace, name } = props;
+  const { signInStatus } = useStore(authStore);
+  const [featuredWorkspaces, setFeaturedWorkspaces] = useState<{ name: string; namespace: string }[]>();
+
+  const isAuthInitialized = signInStatus !== 'uninitialized';
+
+  useEffect(() => {
+    const fetchData = async () => {
+      setFeaturedWorkspaces(await Ajax().FirecloudBucket.getFeaturedWorkspaces());
+    };
+    if (signInStatus === 'signedOut') {
+      fetchData();
+    }
+  }, [signInStatus]);
+
+  const isFeaturedWorkspace = () => _.some((ws) => ws.namespace === namespace && ws.name === name, featuredWorkspaces);
+
+  return cond(
+    [
+      !isAuthInitialized || (signInStatus === 'signedOut' && featuredWorkspaces === undefined),
+      () => centeredSpinner({ style: { position: 'fixed' } }),
+    ],
+    [signInStatus === 'signedOut' && isFeaturedWorkspace(), () => h(DashboardPublic, props)],
+    [signInStatus === 'signedOut', () => h(SignIn)],
+    () => h(WorkspaceDashboard, props)
+  );
+};
+
+interface WorkspaceDashboardProps {
+  namespace: string;
+  name: string;
+}
+
+interface WorkspaceDashboardComponentProps extends WorkspaceDashboardProps {
+  refreshWorkspace: () => void;
+  storageDetails: StorageDetails;
+  workspace: Workspace;
+}
+
+const WorkspaceDashboardComponent = (
```

Comment:
Super nit: the component is returned by `forwardRef`. The argument to `forwardRef` is a render function.

```ts
const WorkspaceDashboardForwardRefRenderFunction = (
```